spacy/pipeline/attributeruler.py

```diff
@@ -0,0 +1,252 @@
+import srsly
+from typing import List, Dict, Union, Iterable, Any
+from pathlib import Path
+
+from .pipe import Pipe
+from ..errors import Errors
+from ..language import Language
+from ..matcher import Matcher
+from ..symbols import IDS
+from ..tokens import Doc, Span
+from ..tokens._retokenize import normalize_token_attrs, set_token_attrs
+from ..vocab import Vocab
+from .. import util
+
+
+MatcherPatternType = List[Dict[Union[int, str], Any]]
+
+
+@Language.factory(
+    "attribute_ruler",
+    assigns=[],
+    default_config={},
+    scores=[],
+    default_score_weights={},
```

Comment:
```suggestion
```

Possible nitpick, but if the component isn't setting any of those, we might as well not set them here?